authentication/pipeline/user.py

```diff
@@ -108,29 +108,27 @@ def create_user_via_email(
             errors=["Full name must be at least 2 characters long."],
         )
 
-    data["email"] = kwargs.get("email", kwargs.get("details", {}).get("email"))
+    data["email"] = (
+        kwargs.get("email", kwargs.get("details", {}).get("email")).strip().lower()
+    )
+
+    if User.objects.filter(email__iexact=data["email"]).exists():
+        raise InvalidEmail(backend)
```

Comment:
What would happen if we throw `UnexpectedExistingUserException` here instead of `InvalidEmail` exception, I feel like throwing InvalidEmail might be a little too general because the email is valid & passes the validation but the problem is that it already exists.

Reply:
The `InvalidEmail` exception is being handled here. https://github.com/mitodl/mitxpro/blob/333ab778a77d16d2cb4dc655a7fe66d698d49905/authentication/serializers.py#L92 It returns the `existing-user` state in the API on the other hand `UnexpectedExistingUserException` returns the `error` state in the API and there is no error message. Here are screenshots where I raised `UnexpectedExistingUserException`.
Backend:
<img width="563" alt="Screenshot 2023-05-12 at 4 21 25 PM" src="https://github.com/mitodl/mitxpro/assets/77275478/6507631c-f2e9-4f62-a34e-fddc2479b73c">
Frontend:
<img width="631" alt="Screenshot 2023-05-12 at 4 26 41 PM" src="https://github.com/mitodl/mitxpro/assets/77275478/bc17127f-6ea6-46d8-8499-64ce00c14abc">